### PR TITLE
Added 50% zoom option in darkroom

### PR DESCRIPTION
**Pull request by mondoman712**
As requested in [issue 12155](https://redmine.darktable.org/issues/12155), I posted the patch there but thought I'd try it as a PR here as well.

### CODE REVIEW

File: src/libs/navigation.c [hunk 1]
```diff
@@ -481,6 +486,10 @@ static void _zoom_preset_2(GtkButton *button, gpointer user_data)
 {
   _zoom_preset_change(3);
 }
+static void _zoom_preset_3(GtkButton *button, gpointer user_data)
```
Comment (rabauke) [inline src/libs/navigation.c]: I would like to suggest to rename the functions  ``_zoom_preset_1`` to ``_zoom_preset_3`` to something more significant, e.g., ``_zoom_preset_100percent`` and so on.

File: src/libs/navigation.c [hunk 2]
```diff
@@ -454,6 +454,11 @@ static void _zoom_preset_change(int val)
     scale = 2.0f;
     zoom = DT_ZOOM_FREE;
   }
+  else if(val == 4)
```
Comment (rabauke) [inline src/libs/navigation.c]: This long chain of `if-else-if` statements may be replaced by a `case` statement.

**Comment by mondoman712** [comment]
@rabauke Thanks for the comments, I've made the changes suggested.

**Comment by TurboGit** [comment]
If I'm not mistaken this is now implemented in dt, right?

**Comment by mondoman712** [comment]
@TurboGit The patch I posted to [redmine](https://redmine.darktable.org/issues/12155#change-35520) got added but this includes a couple extra changes that just clean up the code a bit thanks to @rabauke 's suggestions.

**Comment by TurboGit** [comment]
Ok, so please can you rebase this to current master and resolve the conflicts? Thanks.

**Comment by mondoman712** [comment]
So on closer inspection there's bin another commit since I wrote this that changes a few things in the same area so the only thing left here is changing an if to a case so I'm just going to close this PR.

**Pull request closed without merging**